.github/workflows/release.yml

```diff
@@ -0,0 +1,40 @@
+name: Release
+
+# This workflow follows Pypi guidelines for publishing using trusted publishers
+# See https://docs.pypi.org/trusted-publishers/using-a-publisher/ for more details.
+
+on:
+  release:
+    types: [published]
+
+jobs:
+  package:
+    name: Package
+    uses: ./.github/workflows/package.yml
+
+  release:
+    name: Release
+    needs: [package]
+    runs-on: ubuntu-latest
+
+    # Make the GH environment explicit
+    environment: release
+
+    # Mandatory for trusted publishing
+    permissions:
+      contents: write
+      id-token: write
+
+    steps:
+    - name: Download build artifacts
+      uses: actions/download-artifact@v4
+      with:
+        path: dist/
+
+    - name: Upload release assets
+      uses: softprops/action-gh-release@v2
+      with:
+        files: dist/*.whl
+
+    - name: Publish package to PyPI
```

Comment:
This last step depends only on the "Download" step, correct? Not on the gh-release step. It wasn't clear to me until I checked the documentation so maybe state it in a comment?

Actually, wait: shouldn't `gh-release` be last, _after_ `pypi-publish`? What is its purpose, just a "backup" of the PyPI release, isn't it? In case of a failure/glitch, I think it's better to have the official PyPI release published without the backup rather than the opposite?

Reply:
It's the same file but the gh-release step attaches it to the GitHub release, instead of uploading it manually.

Reply:
I think you missed my point. I'll submit a follow-up PR.

Reply:
>  In case of a failure/glitch, I think it's better to have the official PyPI release published without the backup rather than the opposite?

In case of a failure/glitch, you can retrigger the workflow to ensure that the full release workflow completes.